### DISCUSSION TITLE
API Endpoints for `Fork pull request workflows from outside collaborators` Option?

### DISCUSSION THREAD

**Discussion by ajschmidt8** [community/community#35808] [Actions]
I am looking for some API endpoints to `GET` / `SET` the `Fork pull request workflows from outside collaborators` option shown in the screenshot below for both an organization and a repository. I thought they'd be in https://docs.github.com/en/rest/actions/permissions, but I don't see them there.

Does anyone know if such endpoints exist?

If not, are there plans to make them available?

---

![image](https://user-images.githubusercontent.com/7400326/195155093-86e4e3b0-bb47-4b23-99c1-a13ca81cf859.png)

**Comment by Garcel** [comment]
Just reached this. Any update on this?

**Comment by cp-endor** [comment]
is there a way to get these settings via API ??

**Comment by JAORMX** [comment]
Having such a setting would be very handy

**Comment by arunsathiya** [comment]
I scanned [permissions](https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28) and [workflow runs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28) today, and it doesn't look like it's possible yet.

**Comment by Claire-Thib** [comment]
I also would love this feature. Has there been an update?

**Comment by Kamilcuk** [comment]
+1

**Comment by crabique** [comment]
As I understand, this feature was originally released in [October of 2021](https://github.blog/open-source/maintainers/github-actions-update-helping-maintainers-combat-bad-actors/), yet it still only exists in the Web UI? 🤔

**Comment by ajschmidt8** [comment]
These endpoints seem to be available now based on https://github.blog/changelog/2025-07-31-github-actions-new-apis-and-windows-latest-migration-notice/.

Endpoints are below:

- https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#get-fork-pr-contributor-approval-permissions-for-an-organization
- https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#set-fork-pr-contributor-approval-permissions-for-an-organization
- https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#get-fork-pr-contributor-approval-permissions-for-a-repository
- https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#set-fork-pr-contributor-approval-permissions-for-a-repository

Enterprise endpoints:

- https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions?apiVersion=2022-11-28#get-fork-pr-contributor-approval-permissions-for-an-enterprise
- https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions?apiVersion=2022-11-28#set-fork-pr-contributor-approval-permissions-for-an-enterprise
